pom.xml

```diff
@@ -0,0 +1,232 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+	This is a template for a Reactome Maven based project.
+	Please customize this template by filling in values in square brackets.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+
+	<groupId>org.reactome.[optionalsubdomain]</groupId>
+	<artifactId>[lower-case-name-using-kebab-case]</artifactId>
+	<version>1.0-SNAPSHOT</version>
+	<packaging>jar</packaging>
+
+	<name>[name-of-project]</name>
+	<description>[description of project]</description>
+	<url>https://reactome.org</url>
+	<licenses>
+		<license>
+			<name>[license name]</name>
+			<url>[license url]</url>
+		</license>
+	</licenses>
+
+	<developers>
+		<developer>
+			<name>[Your Name]</name>
+			<email>[your@workemail.com]</email>
+			<organization>Reactome</organization>
+			<organizationUrl>https://reactome.org</organizationUrl>
+		</developer>
+	</developers>
+
+	<scm>
+		<connection>scm:git:git://github.com/reactome/[name-of-repository].git</connection>
+		<developerConnection>scm:git:ssh://github.com/reactome/[name-of-repository].git</developerConnection>
+		<url>https://github.com/reactome/[name-of-repository]</url>
+		<tag>develop</tag>
+	</scm>
+
+	<properties>
+		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+
+		<!--
+		integration tests run by default when running the command "mvn verify"; can be overridden using the option
+		"-DskipITs=true"
+		-->
+		<skipITs>false</skipITs>
+
+		<!--
+		make a property for any dependency versions used more than once
+		-->
+		<junit.version>5.5.2</junit.version>
+		<log4j.version>2.13.0</log4j.version>
+	</properties>
+
+	<dependencies>
+		<!--
+		Main dependencies
+			- List dependencies needed by the main code base in this section
+		-->
+		<dependency>
+			<groupId>org.apache.logging.log4j</groupId>
+			<artifactId>log4j-api</artifactId>
+			<version>${log4j.version}</version>
+		</dependency>
+		<dependency>
+			<groupId>org.apache.logging.log4j</groupId>
+			<artifactId>log4j-core</artifactId>
+			<version>${log4j.version}</version>
+		</dependency>
+
+		<!--
+		Test dependencies
+			- List dependencies needed ONLY by tests in this section
+		-->
+		<dependency>
+			<groupId>org.hamcrest</groupId>
+			<artifactId>hamcrest</artifactId>
+			<version>2.2</version>
```

Comment:
I thought hamcrest was automatically included by junit. Why is it mentioned here?

Reply:
It looks like it is needed - I tried removing the dependency and Hamcrest's imports couldn't be found.

Reply:
Where is hamcrest needed?

Reply:
I created a default test that uses it in `src/test/java/TestMain.java`.  I'd like to encourage using Hamcrest.  The assert statements are usually more readable and provide good information on a test failing.

Reply:
Ok. But, I think that sort of dependency goes a bit beyond the scope of what a template project should be. Mind you, since this is a _Reactome_ template, the reactome-base project probably _should_ be listed as a dependency.

Reply:
I was thinking I may add the reactome-base project as an optional dependency.  It's definitely essential for projects using the Java API for connecting to a MySQL database, but it may not be needed if the project is graph db based.

I was thinking I could add it as a dependency but commented out with an explanation of enabling it if the Java API for accessing MySQL databases is needed.

Perhaps I could do the same with the Hamcrest dependency.

Reply:
I think graph-core (https://github.com/reactome/graph-core) is the essential dependency for projects that use the graph database. So you could list _both_ as optional dependencies.

Reply:
Yeah, that's a good idea.  I'll include explanations for when each would be used along with them.

Reply:
actually, it may be better to just ask ;)

@gsviteri:  which Reactome dependency do all graphdb-related projects require?